index.html

```diff
@@ -5,31 +5,53 @@
     <link rel="stylesheet" href="index.css">
     <link href="https://fonts.googleapis.com/css?family=Satisfy" rel="stylesheet">
     <script src="https://sdk.amazonaws.com/js/aws-sdk-2.145.0.min.js"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/Chart.js/2.7.1/Chart.bundle.js"></script>
     <script type='text/javascript' src='hue.js'></script>
     <script type='text/javascript' src='aws.js'></script>
     <script type='text/javascript' src='Microphone.js'></script>
+    <script type='text/javascript' src='vals-chart.js'></script>
 
   </head>
-    <div class="header">
-      <h1>Welcome to Party Mode</h1>
-      <h2>We'll kindly ask for access to your microphone before we begin the show.</h2>
-      <p>Please enter your hue API.<span><a href="https://www.meethue.com/api/nupnp" target="_blank">Get it!</a></span></p>
-      <input id="hueIP" type="text" name="firstname" placeholder="000.000.000.">
-    <div>
+    <body>
+      <div class="content-wrapper">
+        <div class="header">
+          <h1>Party Mode</h1>
+          <h2>We'll kindly ask for access to your microphone before we begin the show.</h2>
+          <p>Please enter your hue API. <span><a href="https://www.meethue.com/api/nupnp" target="_blank"> Get it!</a></span></p>
+          <input id="hueIP" type="text" name="firstname" placeholder="000.000.000.">
+        </div>
 
+        <div id="user-values">
+          <h3>Your Values</h3>
+
+          <p>
+            <span id="your-volume"></span>
+            <span id="your-highs"></span>
+            <span id="your-mids"></span>
+            <span id="your-bass"></span>
+          </p>
+        </div>
+
+        <div class="chart-wrapper">
+          <canvas id="myChart"></canvas>
+        </div>
+      </div>
+    </div>
 
-  <body>
     <script>
     var Mic = new Microphone();
 
+    allData = []
+    getS3Files();
```

Comment:
It's not immediately apparent how/when allData gets populated. It'd be preferable to have getS3Files take a callback that then sets allData (or pushes into allData if you'd rather not overwrite what's already there.

```
getS3Files(function(s3Data) {
  allData = s3Data;
});
```

allData is also a bit ambiguous. `s3Data` may be more apropos.

Reply:
I definitely should do this.